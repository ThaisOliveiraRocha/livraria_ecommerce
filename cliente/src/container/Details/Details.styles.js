import styled from "styled-components";

export const Body = styled.div`
  width: 100%;
  height: 100%;
  top: 100px;
  margin-bottom: 100px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
