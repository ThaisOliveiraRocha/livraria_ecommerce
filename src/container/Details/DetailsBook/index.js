import DetailsBook from './DetailsBook';

export default DetailsBook;
