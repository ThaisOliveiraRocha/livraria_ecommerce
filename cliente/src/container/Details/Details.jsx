import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { Body } from "./Details.styles";
import { connect } from "react-redux";
import Layout from "../../components/Layout";
import DetailsBook from "./components/DetailsBook";
import ModalComponent from "../../components/ModalComponent";
import { removeLivro, showModal } from "../../store/actions";
import { withRouter } from "react-router-dom";
import SinopseBook from "./components/SinopseBook";

export const Details = ({
  detalhes,
  history,
  funcaoConfirma,
  cancel,
  showModal
}) => {
  if (Object.keys(detalhes).length === 0) {
    {
      history.push("/home");
    }
    return null;
  }
  return (
    <Layout>
      {showModal && (
        <ModalComponent
          temaModal="Deseja excluir o item?"
          mensagem="O item selecionado será excluído permanentemente."
          onClose={cancel}
          funcaoConfirma={funcaoConfirma}
          cancel={cancel}
        />
      )}
      <Body>
        <DetailsBook livro={detalhes} />
        <SinopseBook livro={detalhes} />
      </Body>
    </Layout>
  );
};

Details.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = state => ({
  detalhes: state.livraria.detalhes,
  showModal: state.livraria.removeModal
});

const mapDispatchToProps = dispatch => ({
  funcaoConfirma: () => {
    dispatch(removeLivro());
  },
  cancel: () => {
    dispatch(showModal("", false));
  }
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Details)
);
