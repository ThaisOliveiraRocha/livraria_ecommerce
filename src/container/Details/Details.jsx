import React, { useEffect } from "react";
import PropTypes from "prop-types";
//import { Test } from './Details.styles';
import { connect } from "react-redux";
import Layout from "../../components/Layout";
import DetailsBook from "./components/DetailsBook";
import ModalComponent from "../../components/ModalComponent";
import { removeLivro, showModal } from "../../store/actions";


const Details = ({ detalhes, history, funcaoConfirma, cancel, showModal }) => {
  if (Object.keys(detalhes).length === 0) {
    {
      history.push("/");
    }
    return null;
  }
  return (
    <Layout>
      {showModal && (
      <ModalComponent
        temaModal="Deseja excluir o item?"
        mensagem="O item selecionado será excluído permanentemente."
        onClose={cancel}
        funcaoConfirma={funcaoConfirma}
        cancel={cancel}
      />
    )}
      <DetailsBook livro={detalhes} />
    </Layout>
  );
};

Details.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = state => ({
  detalhes: state.livraria.detalhes,
  showModal: state.livraria.removeModal
});

const mapDispatchToProps = dispatch => ({
  funcaoConfirma: () => {
    dispatch(removeLivro());
  },
  cancel: () => {
    dispatch(showModal("", false));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Details);
