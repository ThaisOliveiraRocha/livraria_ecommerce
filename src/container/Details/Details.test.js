import React from "react";
import { shallow } from "enzyme";
import Details from "./Details";

describe("<Details />", () => {
  it("should render without crashing", () => {
    shallow(<Details />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<Details />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
