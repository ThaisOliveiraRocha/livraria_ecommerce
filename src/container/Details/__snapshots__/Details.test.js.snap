// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should render basic infos 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
